Port `argmax` to structured kernels.

ysiraichi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59938 Port `argmin` kernel to structured kernels.
* **#59937 Port `argmax` to structured kernels.**
* #59373 Using meta checks for unary `torch.all` and `torch.any`.
* #59372 Port `any` kernel to structured kernels.
* #59371 Port `all` kernel to structured kernels.

Tracking issue: #55070

Differential Revision: [D29104397](https://our.internmc.facebook.com/intern/diff/D29104397)

ysiraichi committed:
Port `argmax` to structured kernels.

Tracking issue: #55070

[ghstack-poisoned]

ysiraichi committed:
Update on "Port `argmax` to structured kernels."


Tracking issue: #55070

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29104397).

ysiraichi committed:
Update on "Port `argmax` to structured kernels."


Tracking issue: #55070

Differential Revision: [D29104397](https://our.internmc.facebook.com/intern/diff/D29104397)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29104397).
